pkg/util/async/bounded_frequency_runner.go

```diff
@@ -199,6 +200,17 @@ func (bfr *BoundedFrequencyRunner) Run() {
 	}
 }
 
+// RetryAfter ensures that the function will run again after no later than interval. (It
+// essentially temporarily changes the BoundedFrequencyRunner's maxInterval.) RetryAfter
+// can ONLY be called from inside a run of the BoundedFrequencyRunner's function. Its
+// effects are undefined if called at any other time.
```

Comment:
@danwinship is there any way to guarantee this, or to error out if it's not called from inside tryRun()?

Reply:
You could do something hacky like checking `runtime.Callers()`...

In particular, since go doesn't have "thread IDs" or the like, there's no nice way to distinguish "was called from `bfr.tryRun`" from "was called while `bfr.tryRun` was running in another thread".